### PR TITLE
Better error message when constructing a motor controller group

### LINKED ISSUE #714: [BUG]: bad error message for SpeedControllerGroup construction
### Problem description

The error message generated is useless, we should fix it.

### Operating System

Windows, MacOS, Linux, RoboRIO

### Installed Python Packages

_No response_

### Reproducible example code

```text
>>> import wpilib
>>> wpilib.SpeedControllerGroup(1,2,3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Unable to cast Python instance to C++ type (compile in debug mode for details)
```

### PR DESCRIPTION
**Pull request by virtuald**
- Fixes #714

### CODE REVIEW

File: gen/MotorControllerGroup.yml [hunk 1]
```diff
@@ -16,12 +16,18 @@ classes:
...
+            std::vector<std::shared_ptr<MotorController>> v;
+            v.reserve(args.size());
+            int i = 1;
+            for (auto &arg: args) {
+              try {
+                auto mc = arg.cast<std::shared_ptr<MotorController>>();
+                if (!mc) throw py::cast_error();
+                v.push_back(mc);
+              } catch (py::cast_error &e) {
+                throw py::value_error(py::str("Argument {} must be a MotorController (got '{}')").format(i, py::repr(arg)));
```
Comment (auscompgeek) [inline gen/MotorControllerGroup.yml:28]:
```suggestion
                throw py::type_error(py::str("Argument {} must be a MotorController (got '{}')").format(i, py::repr(arg)));
```

File: tests/test_wpilib.py [hunk 2]
```diff
@@ -17,3 +20,24 @@ def test_smart_dashboard_putdata():
...
+    assert t2.get() == pytest.approx(1)
+
+    g.set(-1)
+    assert t1.get() == pytest.approx(-1)
+    assert t2.get() == pytest.approx(-1)
+
+
+def test_motorcontrollergroup_error():
+    with pytest.raises(
+        ValueError, match=re.escape("Argument 1 must be a MotorController (got '1')")
```
Comment (auscompgeek) [inline tests/test_wpilib.py:41]:
```suggestion
        TypeError, match=re.escape("Argument 1 must be a MotorController (got '1')")
```

File: gen/MotorControllerGroup.yml [hunk 3]
```diff
@@ -16,12 +16,18 @@ classes:
...
               throw py::value_error("requires 1+ arguments");
             }
-            using Vector = std::vector<std::shared_ptr<MotorController>>;
-            Vector v = args.cast<Vector>();
-            for (auto iter = v.cbegin(); iter != v.cend(); ++iter) {
-              if (!*iter) {
-                throw py::value_error("Arguments must not be None");
+            std::vector<std::shared_ptr<MotorController>> v;
+            v.reserve(args.size());
+            int i = 1;
```
Comment (auscompgeek) [inline gen/MotorControllerGroup.yml:21]: Should this count from 0 instead?

**Review by auscompgeek — APPROVED** [review]
LGTM with comments

File: gen/MotorControllerGroup.yml (hunk 3 above)
Comment (virtuald) [reply gen/MotorControllerGroup.yml:21]: I feel like most people count arguments from 1? Particularly when speaking. "first parameter, second parameter..."

Comment (auscompgeek) [reply gen/MotorControllerGroup.yml:21]: Fair enough. I wonder whether any stdlib functions have a similar error message.

**Pull request merged**